lib/libvarnishapi/vsl_dispatch.c

```diff
@@ -1274,15 +1274,54 @@ vslq_shmref_check(struct VSLQ *vslq)
 	return (vsl_more);
 }
 
+static unsigned
+vslq_candidate(struct VSLQ *vslq, const uint32_t *ptr)
+{
+	enum VSL_transaction_e type;
+	enum VSL_reason_e reason;
+	struct VSL_data *vsl;
+	enum VSL_tag_e tag;
+	unsigned p_vxid;
+	int i;
+
+	CHECK_OBJ_NOTNULL(vslq, VSLQ_MAGIC);
+	AN(ptr);
+
+	assert(vslq->grouping != VSL_g_raw);
+	if (vslq->grouping != VSL_g_vxid)
+		return (1);
+
+	vsl = vslq->vsl;
+	CHECK_OBJ_NOTNULL(vsl, VSL_MAGIC);
+	if (!vsl->c_opt && !vsl->b_opt)
+		return (1);
+
+	tag = VSL_TAG(ptr);
+	assert(tag == SLT_Begin);
+	i = vtx_parse_link(VSL_CDATA(ptr), &type, &p_vxid, &reason);
+
+	if (i != 3 || type == VSL_t_unknown)
+		return (0);
+	if (vsl->c_opt && !vsl->b_opt && !VSL_CLIENT(ptr))
+		return (0);
+	if (vsl->b_opt && !vsl->c_opt && !VSL_BACKEND(ptr))
+		return (0);
+	if (type == VSL_t_sess)
```

Comment:
I do not understand why we can do this. Why can we just drop the session transactions here?

Reply:
Because we know that we have at least one of `c_opt` or `b_opt` set, and sessions show up for neither.

Reply:
side note: as the sessions might live for a long time, is the issue maybe more about the fact that many "client" queries are not interested in the session info (but rather only requests)? Could it make more sense to give sessions their own option on the same level as b, c and

Reply:
thank you.